screens/common/GenreList.js

```diff
@@ -0,0 +1,297 @@
+import React, { useEffect, useState } from 'react';
+import { View, StyleSheet, FlatList, Dimensions } from 'react-native';
+import { ActivityIndicator } from 'react-native-paper';
+import { TabBar, TabView } from 'react-native-tab-view';
+import { MOVIE_DETAIL, PERSON_DETAIL, TV_SHOW_DETAIL } from '../../navigators/NavigatorNames';
+import { createUrl, GET, GET_ALL } from '../../network/Api';
+import { API_DISCOVER_MOVIES, API_DISCOVER_TV, API_MOVIES_GENRES, API_MOVIE_CREDITS, API_TV_GENRES, PARAM_LANGUAGE, PARAM_LANGUAGE_VALUE, PARAM_PAGE, PARAM_WITH_GENRES, VAR_MOVIE_ID } from '../../network/NetworkData';
+import { colorAccent, colorPrimary, colorPrimaryDark } from '../../utils/colors';
+import { globalStyles } from '../../utils/globalStyles';
+import { distinctArray } from '../../utils/ValueUtils';
+import MovieListItem from '../movies/MovieListItem';
+import PersonListItem from '../person/PersonListItem';
+
+const GenreList = ({ route, navigation }) => {
+    const { genreId, genreName } = route.params;
+
+    const [isLoadingMovies, setLoadingMovies] = useState(true);
+    const [genreMovies, setGenreMovies] = useState([]);
+    const [movies, setMovies] = useState([]);
+    const [currentPageMovies, setCurrentPageMovies] = useState(1);
+    const [totalPageMovies, setTotalPageMovies] = useState(null);
+    const [paginationRunningMovies, setPaginationRunningMovies] = useState(false);
+
+    const [isLoadingTV, setLoadingTV] = useState(true);
+    const [genreTV, setGenreTV] = useState([]);
+    const [tvShows, setTVShows] = useState([]);
+    const [currentPageTV, setCurrentPageTV] = useState(1);
+    const [totalPageTV, setTotalPageTV] = useState(null);
+    const [paginationRunningTV, setPaginationRunningTV] = useState(false);
+
```

Comment:
tooo many state variables. Is it possible to create a javascript class or object ( with different properties ) and use it in a single state variable?